### ISSUE TITLE
Automatically remove uninstalled games

### ISSUE THREAD

**Issue by JosefNemec** [JosefNemec/Playnite#826]
Original request: https://www.reddit.com/r/playnite/comments/9wxauj/can_i_automatically_remove_uninstalled/

I don't think we should do this automatically, but add some menu function to do it on demand instead.

**Comment by Pralom** [comment]
Maybe if it warns you that the directory for this game isn't found and a pop up appears asking you to remove it, it will be better.

**Comment by kllr007** [comment]
Yeah this is annoying. Playnite now showing games that are not installed and it is confusing.

**Comment by T-0pel** [comment]
@JosefNemec I could take a look at adding this as a menu function that can be launched manually. It would check whether the game .exe file exists. If it does not, it will remove the game from the library. I think it should only target games that were not added automatically by the plugins. Please let me know what you think.

**Comment by JosefNemec** [comment]
@T-0pel The original request was for emulated games specifically, so just checking exe is not enough. You would have to make separate checks for exe for standard games and rom check for emulated ones, which shouldn't be a problem.

It should definitely be added as separate menu function and only executed on demand. Also we should show the list of games that will be removed and ask for configuration to make sure we don't remove something by accident. Maybe even add an option to deselect game from the removal.

And yes it should only work with games that were not added via library plugin (since they would be re-imported back anyways).

**Comment by matdombrock** [comment]
I would love to see this implemented if possible.

**Comment by ElTioRata** [comment]
This is annoying me, please implement this.

**Comment by Ansem93** [comment]
Any hope to see this feature implemented? 
I think it should be an option to have uninstalled games automatically removed or at least a manual check. It's incredibly annoying to have the uninstalled game still in the home page

**Comment by reloxx13** [comment]
+1

**Comment by NineKain** [comment]
+1

**Comment by OxygenEgg** [comment]
+1

**Comment by Getroekah** [comment]
+1
This would save me so much headaches

**Comment by apotenza92** [comment]
I think this should be considered as a user preference that's on by default. 
If something is uninstalled via any means - whether Playnite or directly via Steam etc. - Playnite should have a way to remove that game from the user's library automatically during the next game library update. 

I personally think this should be on by default as it's confusing for new users to see uninstalled games still visible by default in Playnite.

**Comment by Pal1ndrome** [comment]
I uninstalled several games from Xbox and Steam, refreshed my Playnite library (also closed and reopened), and the games still show up. Tried with and without the "Installed" filter.
